arm-gcc.cmake

```
@@ -1,45 +1,58 @@
 set(CMAKE_SYSTEM_NAME Linux)
 set(CMAKE_SYSTEM_PROCESSOR ARM)
-
-if(MINGW OR CYGWIN OR WIN32)
-    set(UTIL_SEARCH_CMD where)
-elseif(UNIX OR APPLE)
-    set(UTIL_SEARCH_CMD which)
-endif()
+set(CMAKE_CROSSCOMPILING TRUE)
 
 set(TOOLCHAIN_PREFIX arm-none-eabi-)
 
-execute_process(
-  COMMAND ${UTIL_SEARCH_CMD} ${TOOLCHAIN_PREFIX}gcc
-  OUTPUT_VARIABLE BINUTILS_PATH
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-# Use only the topmost path, if more than one is found
-string(REGEX REPLACE "\n.+" "" BINUTILS_PATH "${BINUTILS_PATH}")
+if(EXISTS "${ARMGCC_TOOLCHAIN_PATH}/${TOOLCHAIN_PREFIX}gcc")
```

When `ARMGCC_TOOLCHAIN_PATH` is explicitly set but does not point to a valid arm compiler, this would silently ignore it and just use the default compiler in the `PATH`. I believe it would be more logical/intuitive to throw an error and abort.

---

Good point. If set and this does not produce valid settings an error is warranted.